### PR TITLE
[libc][c23][fenv] Implement fetestexceptflag

### LINKED ISSUE #87565: [libc][c23][fenv] implement fetestexceptflag
See `7.6.4.6 The fetestexceptflag function`.  Another C23 addition. We have fetestexcept, but not the symmetrical API for accepting a `fexcept_t*`.

### PR DESCRIPTION
**Pull request by robincaloudis**
Provide C23 `fetestexceptflag` function according to 7.6.4.6 in the latest [revision of the C standard](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3096.pdf) from 2023-04-02.

Closes https://github.com/llvm/llvm-project/issues/87565.

### CODE REVIEW

File: libc/test/src/fenv/exception_flags_test.cpp
Comment (nickdesaulniers) [inline libc/test/src/fenv/exception_flags_test.cpp:19]: s/GetExceptFlagAndSetExceptFlagAndTestExceptFlag/GetSetTestExceptFlag/

File: libc/test/src/fenv/exception_flags_test.cpp
Comment (nickdesaulniers) [inline libc/test/src/fenv/exception_flags_test.cpp:2]: Unittests for fe{get|set|test}exceptflag

**Review by nickdesaulniers — COMMENTED** [review]
Do you mind regenerating the docs for fenv.h?

`./libc/utils/docgen/docgen.py fenv.h > libc/docs/fenv.rst`

And thanks for the patch!

File: libc/src/fenv/fetestexceptflag.h
Comment (lntue) [inline libc/src/fenv/fetestexceptflag.h:12]: Probably you can wait for https://github.com/llvm/llvm-project/pull/88467 to use `hdr/fexcept_t.h` instead.

**Comment by robincaloudis** [comment]
Thanks for the review, @nickdesaulniers and @lntue. I fully agree on the things you pointed out and made the corresponding changes. As https://github.com/llvm/llvm-project/pull/88467 is already approved, let's wait until it is merged and use `#include "hdr/types/fexcept_t.h"` instead of `#include <fenv.h>`.

**Comment by robincaloudis** [comment]
All done. Do you mind to take another look? Thanks!

File: libc/test/src/fenv/exception_flags_test.cpp
Comment (nickdesaulniers) [inline libc/test/src/fenv/exception_flags_test.cpp:58]: Can you add one more `fetestexceptflag` after `fputil::raise_except(FE_OVERFLOW | FE_INEXACT)` that verifies that those 2 are set?

Comment (robincaloudis) [reply libc/test/src/fenv/exception_flags_test.cpp:58]: Good idea!

File: libc/test/src/fenv/exception_flags_test.cpp
Comment (nickdesaulniers) [inline libc/test/src/fenv/exception_flags_test.cpp:68]: Mind adding one more assert:
```suggestion
  ASSERT_EQ(LIBC_NAMESPACE::fetestexceptflag(&overflow_and_inexact_flag,
                                             FE_OVERFLOW | FE_INEXACT),
            FE_OVERFLOW | FE_INEXACT);

```
I think that should work?

Comment (robincaloudis) [reply libc/test/src/fenv/exception_flags_test.cpp:71]: Sure!

**Review by nickdesaulniers — APPROVED** [review]
Will merge once presubmit returns back green! Thanks for the patch!

**Comment by robincaloudis** [comment]
Sounds good. Thank you!

**Pull request merged**
